src/Features/Core/Portable/ExtractMethod/UniqueNameGenerator.cs

```diff
@@ -15,13 +16,25 @@ public UniqueNameGenerator(SemanticModel semanticModel)
             _semanticModel = semanticModel;
         }
 
-        public string CreateUniqueMethodName(SyntaxNode contextNode, string baseName)
+        public string CreateUniqueMethodName(SyntaxNode contextNode, string baseName, bool generateLocalFunction = false)
         {
             Contract.ThrowIfNull(contextNode);
             Contract.ThrowIfNull(baseName);
 
+            if (generateLocalFunction)
+            {
+                // When generating local functions, we also want to take into account the names of local variables.
+                var childNodes = contextNode.ChildNodes().AsArray();
+                if (childNodes.Length > 0)
+                {
+                    // If we take the first child node, we may end up with part of the method header.
+                    // To be safe, we use the last child node, which is guaranteed to be part of the context method's body.
+                    contextNode = (SyntaxNode)childNodes.GetValue(childNodes.Length - 1);
+                }
+            }
+
             return NameGenerator.GenerateUniqueName(baseName, string.Empty,
-               n => _semanticModel.LookupSymbols(contextNode.SpanStart, /*container*/null, n).Length == 0);
+                n => _semanticModel.LookupSymbols(contextNode.SpanStart, container: null, n).Length == 0);
```

Comment:
this should work already.  you're looking up all the symbols within that context-node.  So that should automatically ensure that the local function you want to create will not collide with any other locals in this context-node.

Reply:
I tested this and from what I can tell, when the context node is a MethodDeclarationNode then it will not take into account the names of any locals in the method itself, including any existing local functions.

Reply:
Why is the context node a MethodDeclarationNode?  The idea is that the content node is whre you're going to put the new variable.  So i would expect that the node we would pass in would be more like the node we selected to extract.